lib/module.c

```diff
@@ -75,6 +81,31 @@ struct frrmod_runtime *frrmod_load(const char *spec, const char *dir, char *err,
 	struct frrmod_runtime *rtinfo, **rtinfop;
 	const struct frrmod_info *info;
 
+	char *pErr = NULL;
+	size_t err_remain = 0;
+
+	if (err && (err_len > 0)) {
+		pErr = err;
+		err_remain = err_len - 1;
+	}
+
+#define ERR_RECORD                                                             \
+	do {                                                                   \
+		if (err && err_remain) {                                       \
+			ssize_t used;                                          \
+			used = snprintfrr(pErr, err_remain,                    \
+					  "%sloader error: dlopen(%s): %s",    \
+					  ((pErr != err) ? "\n" : ""),         \
```

Comment:
Syslog messages can't contain `\n`, so the messages generated here would be invalid for syslog.  While we currently do the module loading very early (with `stderr` logging), the option to load things later is something intentionally kept open.  Can we find some better way to report these errors?

Simplest thing I can think of would be to just pass a callback function pointer that frrmod_load invokes for errors.  That would also remove the need for this concatenation construct.

Reply:
FWIW the only reason frrmod_load doesn't directly do a `zlog_err` is that this code was written back when `zlog_err` wouldn't work early in the startup (with no logging configured yet.)  This shouldn't be the case anymore with the logging reworks, though I haven't had the need to actually test it yet…